lib/libutee/include/utee_defines.h

```diff
@@ -75,8 +76,24 @@ static inline uint32_t __tee_alg_get_main_alg(uint32_t algo)
 	/* Bits [11:8] */
 #define TEE_ALG_GET_CHAIN_MODE(algo)    (((algo) >> 8) & 0xF)
 
+/*
```

Comment:
GP has abandoned the structure of the algorithm identifiers and as new algorithms are added this will get worse and worse. With this hack we're able to hang in there for a little longer, but at some point we'll need to bite the bullet and free the code of assumptions on the structure of a algorithm identifier.

Reply:
I agree, but whatever the structure of the identifiers, there will still be characteristics associated with algorithms such as digest function, main algorithm, chaining mode etc. so it might be that this "hack" is just a first step -- we might end up open coding big switch/case statements instead.